src/features/tasks/pages/Task.tsx

```diff
@@ -1,150 +1,59 @@
-    useTrustCommerceSubscriptionPlans(subscriptionPlanId);
-  const { data: billings, isLoading: isLoadingBillings } = useActiveBillings();
-
-  if (isLoadingSubscriptionPlans || isLoadingBillings) {
-    return <Skeleton variant="rectangular" animation="wave" height={280} />;
-  }
-
-  if (!subscriptionPlans || subscriptionPlans.length === 0) {
-    return null;
-  }
-
-  return (
-    <TrustCommerceSubscriptionPayment
-      billings={billings}
-      onSubmit={onSubmit}
-      subscriptionPlan={subscriptionPlans[0]}
-    />
-  );
-};
-
-const SubscriptionPayment = ({
-  carePlanId,
-  patient,
-  task,
-}: {
-  carePlanId?: string;
-  patient: Patient;
-  task: TaskInterface;
-}) => {
-  const { mutate: updateTask } = useUpdateTask();
-  const navigate = useNavigate();
-
-  const isTaskCompleted = task.achievement_status === TaskAchievementStatus.COMPLETED;
-
-  const onSubmit = useCallback(async () => {
-    await updateTask({
-      taskId: task.id,
-      task: {
-        achievement_status: TaskAchievementStatus.COMPLETED,
-      },
-    });
-
-    navigate(`/home/${carePlanId}/tasks`);
-  }, []);
-
-  if (!task.subscriptionPlanId || isTaskCompleted) {
-    return null;
-  }
-
-  if (isStripeEnabled) {
-    return <StripeSubscriptionPaymentWrapper onSubmit={onSubmit} patient={patient} task={task} />;
-  }
-
-  if (isTrustCommerceEnabled) {
-    return (
-      <TrustCommerceSubscriptionPaymentWrapper
-        onSubmit={onSubmit}
-        subscriptionPlanId={task.subscriptionPlanId}
-      />
-    );
-  }
-
-  return null;
-};
+import theme from "styles/theme";
+import useCRMContent from "fetchDataHooks/useCRMContent";
 
 const Task = () => {
   const { desktopEnabled } = useFlags();
+  const addInfoAlert = useContext(AddInfoContext);
   const isMobile = useMediaQuery(theme.breakpoints.down("sm"));
   const { taskId, carePlanId } = useParams<{ taskId: string; carePlanId: string }>();
   // @ts-ignore
   const { data: task, isLoading: taskLoading, isError: taskError, refetch } = useTask(taskId);
   const { isLoading: taskWithCRMLoading, isError: taskWithCRMError } = useCRMContent(task);
+  const [isTaskCompleted, setIsTaskCompleted] = useState<boolean>(
+    task?.achievement_status === TaskAchievementStatus.COMPLETED
+  );
 
-  const isTaskCompleted = task?.achievement_status === TaskAchievementStatus.COMPLETED;
+  useEffect(() => {
+    setIsTaskCompleted(task?.achievement_status === TaskAchievementStatus.COMPLETED);
+  }, [task]);
 
   const { currentPatient } = useCurrentPatient();
   const { mutate: updateTask } = useUpdateTask();
 
+  const onSubmit = async () => {
+    await updateTask({
+      taskId: task.id,
+      task: {
+        achievement_status: TaskAchievementStatus.COMPLETED,
+      },
+    });
+    refetch();
+    addInfoAlert({
+      status: "success",
+      message: "Responses submitted",
+      isToast: true,
+    });
+    setIsTaskCompleted(true);
```

Comment:
If we invalidate the task on update, we should be able to do this
```suggestion
    addInfoAlert({
      status: "success",
      message: "Responses submitted",
      isToast: true,
    });
```

Reply:
same response as above